dump_sqs/dump_sqs.js

```diff
@@ -0,0 +1,142 @@
+const AWS = require('aws-sdk');
+const fs = require('fs');
+const { parseArgs } = require('util');
+
+const args = [
+  { name: "awsProfile", mandatory: true, subcommand: [] },
+  { name: "queueName", mandatory: true, subcommand: [] },
+  { name: "format", mandatory: false, subcommand: [] }
+]
+
+const values = {
+  values: { awsProfile, queueName, format },
+} = parseArgs({
+  options: {
+    awsProfile: {
+      type: "string",
+      short: "a"
+    },
+    queueName: {
+      type: "string",
+      short: "q"
+    },
+    format: {
+      type: "string",
+      short: "f",
+      default: "raw"
+    }
+  },
+});
+
+function _checkingParameters(args, values){
+  const usage = "Usage: node dump_sqs.js --awsProfile <aws-profile> --queueName <queue-name> [--format <output-format>]"
+  //CHECKING PARAMETER
+  args.forEach(el => {
+    if(el.mandatory && !values.values[el.name]){
+      console.log("Param " + el.name + " is not defined")
+      console.log(usage)
+      process.exit(1)
+    }
+  })
+  args.filter(el=> {
+    return el.subcommand.length > 0
+  }).forEach(el => {
+    if(values.values[el.name]) {
+      el.subcommand.forEach(val => {
+        if (!values.values[val]) {
+          console.log("SubParam " + val + " is not defined")
+          console.log(usage)
+          process.exit(1)
+        }
+      })
+    }
+  })
+}
+
+_checkingParameters(args, values)
+
+console.log("Using DLQ Name: " + queueName)
+console.log("Using Format: " + format)
+
+let credentials = null
+var elementsElaborated = []
+
+process.env.AWS_SDK_LOAD_CONFIG=1
+if(awsProfile.indexOf('sso_')>=0){ // sso profile
+  credentials = new AWS.SsoCredentials({profile:awsProfile});
+  AWS.config.credentials = credentials;
+} else { // IAM profile
+  credentials = new AWS.SharedIniFileCredentials({profile: awsProfile});
+  AWS.config.credentials = credentials;
+}
+AWS.config.update({region: 'eu-south-1'});
+
+const sqs = new AWS.SQS();
+
+
+async function _writeInFile(result, filename ) {
+  fs.mkdirSync("result", { recursive: true });
+  fs.writeFileSync('result/' + filename+'.json', JSON.stringify(result, null, 4), 'utf-8')
+}
+
+async function dumpSQS() {
+  const queueParams = {
+    QueueName: queueName
+  };
+  const queueData = await sqs.getQueueUrl(queueParams).promise();
+  let maxNumberOfMessages = queueName.includes(".fifo") ? 1: 10;
+  const params = {
+    QueueUrl: queueData.QueueUrl,
+    MaxNumberOfMessages: maxNumberOfMessages, // Numero massimo di messaggi da recuperare (modificabile) ma deve essere 1 per code FIFO
+    AttributeNames: ['All'],
+    MessageAttributeNames: ['All'],
+    VisibilityTimeout: 20,    // Tempo in secondi di non visibilità del messaggio dalla DLQ
```

Comment:
prova ad aumentare il numero di visibility timeout in base al numero di messaggi in coda.